src/core/net/tcp6.cpp

```diff
@@ -168,17 +168,33 @@ Error Tcp::Endpoint::Bind(const SockAddr &aSockName)
 
 Error Tcp::Endpoint::Connect(const SockAddr &aSockName, uint32_t aFlags)
 {
-    Error               error = kErrorNone;
-    struct tcpcb       &tp    = GetTcb();
-    struct sockaddr_in6 sin6p;
-
-    OT_UNUSED_VARIABLE(aFlags);
+    Error         error         = kErrorNone;
+    struct tcpcb &tp            = GetTcb();
+    uint32_t      fastOpenFlags = aFlags & OT_TCP_CONNECT_NO_FAST_OPEN;
 
     VerifyOrExit(tp.t_state == TCP6S_CLOSED, error = kErrorInvalidState);
 
-    memcpy(&sin6p.sin6_addr, &aSockName.mAddress, sizeof(sin6p.sin6_addr));
-    sin6p.sin6_port = HostSwap16(aSockName.mPort);
-    error           = BsdErrorToOtError(tcp6_usr_connect(&tp, &sin6p));
+    if (fastOpenFlags == OT_TCP_CONNECT_NO_FAST_OPEN)
+    {
+        struct sockaddr_in6 sin6p;
+
+        tp.t_flags &= ~TF_FASTOPEN;
+        memcpy(&sin6p.sin6_addr, &aSockName.mAddress, sizeof(sin6p.sin6_addr));
+        sin6p.sin6_port = HostSwap16(aSockName.mPort);
+        error           = BsdErrorToOtError(tcp6_usr_connect(&tp, &sin6p));
+    }
+    else if (fastOpenFlags == 0)
```

Comment:
Can remove `fastOpenFalgs == 0` check ? and use `else` directly here? 

----

Actually may be more clear if we remove the `fastOpenFalgs` variable and do a direct check in `if`?
```c++
if (aFlags & OT_TCP_CONNECT_NO_FAST_OPEN) 
{
     // ... (code when do not have fast open)
}
else 
{
     // ... fast-open mode
}
```

Reply:
Thanks --- I just pushed an update addressing this.